deepchem/models/tensorgraph/layers.py

```diff
@@ -3112,3 +3113,196 @@ def none_tensors(self):
+      [V, A, mask]
+      V are the vertex features must be of shape (batch, vertex, channel)
+
+      A are the adjacency matrixes for each graph
+        Shape (batch, from_vertex, adj_matrix, to_vertex)
+
+      mask is optional, to be used when not every graph has the
+      same number of vertices
+
+    Returns: tf.tensor
+    Returns a tf.tensor with a graph convolution applied
+    The shape will be (batch, vertex, self.num_filters)
+    """
+    in_tensors = self._get_input_tensors(in_layers)
+    if len(in_tensors) == 3:
+      V, A, mask = in_tensors
+    else:
+      V, A = in_tensors
+      mask = None
+    factors = self.embedding_factors(
+        V, self.num_vertices, name='%s_Factors' % self.name)
+
+    if mask is not None:
+      factors = tf.multiply(factors, mask)
+    factors = self.softmax_factors(factors)
+
+    result = tf.matmul(factors, V, transpose_a=True)
+
+    result_A = tf.reshape(A, (tf.shape(A)[0], -1, tf.shape(A)[-1]))
+    result_A = tf.matmul(result_A, factors)
+    result_A = tf.reshape(result_A, (tf.shape(A)[0], tf.shape(A)[-1], -1))
+    result_A = tf.matmul(factors, result_A, transpose_a=True)
+    result_A = tf.reshape(result_A, (tf.shape(A)[0], self.num_vertices,
+                                     A.get_shape()[2].value, self.num_vertices))
+    # We do not need the mask because every graph has self.num_vertices vertices now
+    if set_tensors:
+      self.out_tensor = result
+    self.out_tensors = [result, result_A]
+    return result, result_A
+
+  def embedding_factors(self, V, no_filters, name="default"):
+    no_features = V.get_shape()[-1].value
+    W = tf.get_variable(
+        '%s_weights' % name, [no_features, no_filters],
+        initializer=tf.truncated_normal_initializer(
+            stddev=1.0 / math.sqrt(no_features)),
+        dtype=tf.float32)
+    b = tf.get_variable(
+        '%s_bias' % self.name, [no_filters],
+        initializer=tf.constant_initializer(0.1),
+        dtype=tf.float32)
+    V_reshape = tf.reshape(V, (-1, no_features))
+    s = tf.slice(tf.shape(V), [0], [len(V.get_shape()) - 1])
+    s = tf.concat([s, tf.stack([no_filters])], 0)
+    result = tf.reshape(tf.matmul(V_reshape, W) + b, s)
+    return result
+
+  def softmax_factors(self, V, axis=1, name=None):
+    max_value = tf.reduce_max(V, axis=axis, keep_dims=True)
+    exp = tf.exp(tf.subtract(V, max_value))
+    prob = tf.div(exp, tf.reduce_sum(exp, axis=axis, keep_dims=True))
+    return prob
+
+  def none_tensors(self):
+    out_tensors, out_tensor = self.out_tensors, self.out_tensor
+    self.out_tensors = None
+    self.out_tensor = None
+    return out_tensors, out_tensor
+
+  def set_tensors(self, tensor):
+    self.out_tensors, self.out_tensor = tensor
+
+
+def GraphCNNPool(num_vertices, **kwargs):
+  gcnnpool_layer = GraphCNNPoolLayer(num_vertices, **kwargs)
+  return [PassThroughLayer(x, in_layers=gcnnpool_layer) for x in range(2)]
+
+
+class GraphCNN(Layer):
+  """
```

Comment:
Could you add a docstring here describing how this layer differs from our other graph conv layers?